lib/open-uri-cached-archive.rb

```diff
@@ -1,5 +1,43 @@
 require 'open-uri-cached-archive/version'
+require 'yaml'
+require 'stringio'
+require 'open-uri'
 
 class OpenUriCachedArchive
-  # Your code goes here...
+  class CachedResponse
+    def initialize(meta_file_path)
+      @meta = YAML.load_file(meta_file_path)
+      @body = File.read(meta_file_path.gsub(/\.meta$/, ''))
+      @status = meta.delete(:status)
+      @base_uri = meta.delete(:base_uri)
+      meta.delete(:content_type)
+    end
+
+    def open_uri_response
+      StringIO.new(body).tap do |response|
+        OpenURI::Meta.init(response)
+        meta.each { |k, v| response.meta_add_field(k, v) }
+        response.status = status
+        response.base_uri = base_uri
+      end
+    end
+
+    private
+
+    attr_reader :meta, :body, :status, :base_uri
+  end
+
+  def initialize(path)
+    @path = path
+  end
+
+  def responses
+    Dir.glob(File.join(path, '**', '*.meta')).map do |meta_file_path|
```

Comment:
If you were to refactor the `body` method out above, then switching to `Pathname.glob` instead of `Dir.glob` here _might_ also help you simplify that. (Though I'm not 100% convinced, because I'm not sure whether there's a super-simple way to go from 'foo/bar.meta' to 'foo/bar' in one shot with a `Pathname`)